### DISCUSSION TITLE
Override Interface for tests

### DISCUSSION THREAD

**Discussion by joelsantos** [hadashiA/VContainer#235] [Q&A]
Hi there. First of all, great project that you have here 👍 I'm in the process of migrating my game from Zenject to it, and I'm having a few issue.
I'm trying to override an interface that is originally registered on the RootLifeScope with a Mock/Substitute, however it's not being picked up correctly. The test has to load a scene, before running.

Right now I have, but this doesn't seem to work.
```cs
public GameSceneTests()
{
    this.gameContainer = Substitute.For<IGameContainer>();
    this.gameContainer.RestoredGame.Returns(true);
            
    LifetimeScope.Create(builder =>
    {
        builder.RegisterInstance(this.gameContainer).As<IGameContainer>();
    });
}
```

Also, it would be great to have a few examples for UnityTests.
Thanks in advance,
Joel Santos

**Comment by hadashiA** [comment]
Hello. Thanks for trying it out !

In my opinion, for each unit test, you can initialize the object manually. (No need for DI magic).

As for how to override the LifetimeScope of an existing scene, the following may be of use.

```csharp
using (LifetimeScope.Enqueue(overrideBuilder => overrideBuilder.RegisterInstance(gameContainer)))
{
    await SceneManager.LoadSceneAsync("TestTarget");

    // or Instantiate LifetimeScope of a test target...
}
```

I hope this helps in some way!

**Reply by JesseTG** [reply]
> In my opinion, for each unit test, you can initialize the object manually. (No need for DI magic).

I disagree. Well, for integration tests, at least. I need to inject a lot of dependencies in my integration tests (not as many as I do for the actual game, though).
